### ISSUE TITLE
New Task: GOOS Pilot adding US data to near-realtime to OBIS in DwC

### ISSUE THREAD

**Issue by MathewBiddle** [ioos/marine_life_data_network#94]
### Who is requesting this?

GOOS

### What is being requested?

SMRU data appropriately QA/QC'ed using ArgosQC, aligned to DarwinCore and shared to OBIS via OTN's IPT.

### What is the requested deadline and why?

2024-10-01

### What is the current status quo (i.e., what happens if this does not get done)?

unclear pathways for data to OBIS

### What indicates this is done (i.e., how do we know this is complete)?

Pilot is pilotted and an implementation plan is built.

### Provide a description or any other important information.

Coordinating with @mmckinzie and @jdpye.

**Comment by MathewBiddle** [comment]
working over at https://github.com/ocean-tracking-network/rt-sat-to-obis

**Comment by mmckinzie** [comment]
I've reached out to ATN researchers as discussed and will get agreed upon data/metadata uploaded next week

**Comment by MathewBiddle** [comment]
Thanks @mmckinzie. I see you added the requested data files and .yml configs to the repository https://github.com/ocean-tracking-network/rt-sat-to-obis. Thanks!

I'm trying to decipher what was added/removed, but reviewing the comparison between main and the commit previous to when you started adding content is a little difficult to trace. See https://github.com/ocean-tracking-network/rt-sat-to-obis/compare/main..5a1d07a3a2790f2452a92fdf8efef7380a0bd0d9

A common tool used in GitHub to clearly identify changes to a repository is using a [Pull Request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests). I can help you work through how and when to use them. It makes it easier for you, as well as the team, to clearly understand what changes have been made. It also has nice integration with issues and notifications. Let's set up a short call to review how to do this.

**Comment by MathewBiddle** [comment]
xref: https://github.com/ocean-tracking-network/rt-sat-to-obis/issues/3

**Comment by jdpye** [comment]
We will harmonize the .yml and config files (which don't yet have a destination standard to conform to, we're making this up as we go!) from here, and we'll endeavour to request all changes happen through PRs from here on out. We're a ways away from the first 'release' of this workflow so i think we'll be OK this time but it's good to get practice in doing things the neat and tidy way!

**Comment by MathewBiddle** [comment]
cc @conniekot

**Comment by mmckinzie** [comment]
I believe Ian J. resumed work on this effort last week. But not sure where things stand.

**Comment by MathewBiddle** [comment]
cc @ianjonsen

**Comment by MathewBiddle** [comment]
from @ianjonsen:

> I’ve hit the first milestone in building out the ArgosQC pkg - I have an automated pipeline working for ATN SMRU tag data.
> https://github.com/ianjonsen/ArgosQC/tree/dev
> To test this, I’ve used the ATN SMRU tag data in https://github.com/ocean-tracking-network/rt-sat-to-obis
> Next milestones are: 
> 1) complete automated QC pipeline for generic WC tag data; 
> 2) complete automated QC pipeline for ATN WC tag data; 
> 3) update pkg documentation; 
> 4) build vignettes

**Comment by mmckinzie** [comment]
I've now received permission from several researchers to share WC SCOUT-TEMP, SCOUT-DSA, and SPLASH-F-391 data and corresponding ATN project/deployment metadata. I hope to get everything migrated over to GitHub early next week. I also had a meeting with WC earlier this week to initiate direct data release to @ianjonsen via the WC API.

**Comment by mmckinzie** [comment]
from @ianjonsen 

The output from the QC would be: 

1) all the relevant original tag files (ie. those where appending QC'd locations make sense) with SSM-estimated locations appended; 

2) the original metadata with QCStartDateTime and QCStopDateTime appended. I can imagine some additional attributes might be useful, like QCMethod - eg. "RArgosQC";  QCVersion - eg. ArgosQC R pkg version number used; QCDateTime - eg. date when QC was last run. 

3) an SSMoutputs.csv file, which contains the SSM-estimated locations & their uncertainty estimates at the QC operator-specified prediction time interval - by default this is 2 h, but could vary by species &/or tag programming.

**Comment by mmckinzie** [comment]
WC SPLASH-F-391, SCOUT-TEMP-361A and SCOUT-DSA data, along with tag deployment and project metadata were added to the ATN input folder and the yml files to the source folder: https://github.com/ocean-tracking-network/rt-sat-to-obis.

Direct data access via WC API has also been requested.

**Comment by MathewBiddle** [comment]
Has this pilot been piloted? 

If so, it would be nice to have a summary of where we stand with a proposal on where to go next. For example, "we shared these data via this pathway and this process should be included in the ATN's data flow (here is how we propose it should work)." Or something like that.

**Comment by ianjonsen** [comment]
I'll let @jdpye add a more detailed response, but we're close. The codebase is just about there for operational use. The following needs to be updated: 

1. final output formatting for an OTN metadata structure; this will be the default for new programs but IMOS & ATN structures are baked in
2. final documentation of complete workflow
3. bug fixes as required - this may include adding capability to handle WC tag data structures not previously encountered

**Comment by laurabrenskelle** [comment]
Is this actively being worked on? Should this still be In Progress? Is it done? Is it blocked by something?
